src/sidebar/services/test/api-test.js

```diff
@@ -17,395 +16,283 @@ const util = require('../../../shared/test/util');
 const routes = require('./api-index.json').links;
 
 describe('sidebar.services.api', function() {
-  let $httpBackend;
-  let $q;
   let fakeAuth;
   let fakeStore;
-  let sandbox;
   let api;
 
-  before(function() {
-    angular.module('h', []).service(
-      'api',
-      proxyquire(
-        '../api',
-        util.noCallThru({
-          angular: angular,
-          '../retry-util': {
-            retryPromiseOperation: function(fn) {
-              return fn();
-            },
-          },
-        })
-      )
-    );
-  });
-
-  beforeEach(function() {
-    sandbox = sinon.sandbox.create();
+  function defaultBodyForStatus(status) {
+    if (status === 204) {
+      return '';
+    } else if (status >= 500) {
+      return '<html><body>Internal Server Error</body></html>';
```

Comment:
See above comment (500s are JSON w/body for `/api` routes in `h`).

Reply:
This is true for 4xx errors, because those all come from h. For 5xx errors those can come from different places (eg. Cloudflare if its DDOS protection were to kick in). So we'd either have to check the content type of the response or just treat it as text.